src/hooks/useCurrentBlockTimestamp.ts

```diff
@@ -1,10 +1,19 @@
 import { BigNumber } from '@ethersproject/bignumber'
+import { useEffect, useState } from 'react'
 
 import { useSingleCallResult } from '../state/multicall/hooks'
 import { useMulticall2Contract } from './useContract'
 
 // gets the current timestamp from the blockchain
 export default function useCurrentBlockTimestamp(): BigNumber | undefined {
+  const [lastBlockTimestamp, setLastBlockTimestamp] = useState<BigNumber | undefined>(undefined)
   const multicall = useMulticall2Contract()
-  return useSingleCallResult(multicall, 'getCurrentBlockTimestamp')?.result?.[0]
+  // TODO:zzmp
+  const currentBlockTimestamp = useSingleCallResult(multicall, 'getCurrentBlockTimestamp')?.result?.[0]
```

Comment:
i'm surprised this is causing issues. Shouldn't `useSingleCassResult` return the same object every time?

also i wonder if there's merit to using `number` over `BigNumber` for primitive equality.

Reply:
I don't know this library's internals. I'll ask someone else to review this to make sure multicall is working correctly, but in my local testing this was returning a new `BigNumber` (with the same value) every time.

Reply:
This appears to be working correctly now. I believe the library was being mis-used.